superset/assets/src/components/AnchorLink.jsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { t } from '@superset-ui/translation';
+
+import URLShortLinkButton from './URLShortLinkButton';
+import getDashboardUrl from '../dashboard/util/getDashboardUrl';
+
+const propTypes = {
+  anchorLinkId: PropTypes.string.isRequired,
+  filters: PropTypes.object,
+  showShortLinkButton: PropTypes.bool,
+  placement: PropTypes.string,
```

Comment:
could you enumerate the allowed types instead of simply `string`? more specific = better.

Reply:
fixed.